enterprise/cypher/physical-planning/src/main/scala/org/neo4j/cypher/internal/compatibility/v3_4/runtime/SlottedRewriter.scala

```diff
@@ -333,17 +326,15 @@ class SlottedRewriter(tokenContext: TokenContext) {
       propExpression
   }
 
-  private def stopAtOtherLogicalPlans(thisPlan: LogicalPlan): (AnyRef) => Boolean = { lp =>
-    lp match {
-      case _: LogicalPlan =>
-        lp != thisPlan
+  private def stopAtOtherLogicalPlans(thisPlan: LogicalPlan): (AnyRef) => Boolean = {
+    case lp@(_: LogicalPlan) =>
+      lp.id != thisPlan.id
```

Comment:
Just a question: Does this assume that we have no collisions in hashCode or does it tolerate that we stop traversing if we have a hash collision?

Reply:
Not sure I follow, what are we hashing?

Reply:
Nevermind. I confused `id` with `debugId`.

Reply:
👍